zeebe/util/src/main/java/io/camunda/zeebe/util/jar/ExternalJarClassLoader.java

```diff
@@ -29,10 +38,25 @@ private ExternalJarClassLoader(final URL[] urls) {
     super(urls);
   }
 
+  /**
+   * Close class loader with verbose log statement.
+   *
+   * <p>Be aware that premature closing may lead to ClassNotFoundException.
+   */
   @Override
   public void close() throws IOException {
-    LOGGER.warn(
-        "Closing the external JAR class loader may lead to ClassNotFoundException; only close if it's truly necessary");
+    close(true);
+  }
+
+  /**
+   * Allows to close the class loader without warning statement
+   *
+   * <p>Be aware that premature closing may lead to ClassNotFoundException.
+   */
+  public void close(final boolean verbose) throws IOException {
+    if (verbose) {
+      LOGGER.warn("Premature closing may lead to ClassNotFoundException");
```

Comment:
:wrench: I would describe the consequence in clearer terms, e.g. "Closing the class loader may cause future calls to load classes to fail with ClassNotFoundException" or something. I know it's what you wrote, but I'm not sure if someone who has little context would understand it the way you phrased it :shrug: Just a suggestion

Reply:
Reworder to `Closing the class loader may cause future class loading calls to fail with ClassNotFoundException`